transform/files_test.go

```diff
@@ -0,0 +1,94 @@
+package transform
+
+import "testing"
+
+func TestIsFileOf(t *testing.T) {
+	types := []kind{company, partner, facility}
+	cases := []struct {
+		name     string
+		expected kind
+	}{
+		{"F.K03200$W.SIMPLES.CSV.D10612.zip", simple},
+		{"F.K03200$Z.D10612.CNAECSV.zip", cnae},
+		{"F.K03200$Z.D10612.MOTICSV.zip", motive},
+		{"F.K03200$Z.D10612.MUNICCSV.zip", city},
+		{"F.K03200$Z.D10612.NATJUCSV.zip", nature},
+		{"F.K03200$Z.D10612.PAISCSV.zip", country},
+		{"F.K03200$Z.D10612.QUALSCSV.zip", qualification},
+		{"K3241.K03200Y0.D10612.EMPRECSV.zip", company},
+		{"K3241.K03200Y0.D10612.ESTABELE.zip", facility},
+		{"K3241.K03200Y0.D10612.SOCIOCSV.zip", partner},
+		{"K3241.K03200Y1.D10612.SOCIOCSV.zip", partner},
+		{"K3241.K03200Y2.D10612.ESTABELE.zip", facility},
+		{"K3241.K03200Y2.D10612.SOCIOCSV.zip", partner},
+		{"K3241.K03200Y5.D10612.EMPRECSV.zip", company},
+	}
+
+	for _, c := range cases {
+		for _, k := range types {
```

Comment:
```suggestion
		t.Run(c.name, func(t *testing.T{
		        for _, k := range types {
			e := c.expected == k
			if r := isFileOf(newDataset(k, "testdata", ""), c.name); r != e {
				t.Errorf("Expected %s to be of kind %s", c.name, c.expected)
			}
		}
		})
```

What do you think of this implementation?

Reply:
Better than mine : ) Changed!